### DISCUSSION TITLE
Scenario "Kihei": Surely Not Another Disk Space Scenario

### DISCUSSION THREAD

**Discussion by fduran** [SadServers/sadservers#32] [Scenarios]
Discussions about scenario [Kihei](https://github.com/fduran/sadservers/tree/main/scenarios/kihei)

**Comment by HarukaMa** [comment]
<details>
<summary>Spoiler</summary>
This is interesting as I cheesed it by making the datafile a sparse one as it's just zero-filled, then I watched replays and found there are two 1 GB unused disks and figured that would be the intended solution.

This could be prevented by filling the datafile with random data and check the content, but I guess it's not needed for fun's sake.

</details>

**Comment by fduran** [comment]
Hello, thanks for your comment. The idea behind the files filled with zeros is to make clear that they can be deleted safely (vs deleting possible production data).

**Reply by HarukaMa** [reply]
Are you referring to a different scenario? the check script explicitly checked that the 5GB file is not deleted and the apparent size has not changed.

**Reply by fduran** [reply]
Ah sorry, yes, was confused with "Manhattan" where you can delete the files.

**Comment by JoeHO888** [comment]
When I tried to strace -f ./kihei, the console terminated immediately. When I opened again, I see below image. is it expected?

![image](https://github.com/user-attachments/assets/e27da124-00eb-4071-a6d8-1ff88786d8aa)

**Reply by fduran** [reply]
Hello @JoeHO888 , this scenario runs with [asciinema](https://asciinema.org/) for the [replay](https://replay.sadservers.com/) feature and it breaks when running `strace` or under some other conditions. We haven't found yet a good solution for this kind of issue.

**Reply by Ismail-Shaker** [reply]
Yeah, this seems to happen sporadically, haven't been able to find a pattern, sometimes it's after running the first strace and other times you can run it 10 times and nothing would happen.
My assumption is that it fills up the disk completely, preventing SSH; you'll need to destroy and recreate the server.
